integration tests fail to delete cleanly

This what I'm seeing on master (8fbc49641a563a30ab71725774bb49d1cfc23b70) now, 
I suspect it did't appear before #513.
```
• Failure [900.007 seconds]
(Integration) Create, Get, Scale & Delete
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:27
  when creating a cluster with 1 node
  /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:53
    and deleting the cluster
    /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:273
      should not return an error [It]
      /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:274

      Timed out after 900.001s.
      Expected process to exit.  It did not.

      /go/src/github.com/weaveworks/eksctl/integration/common_test.go:65
------------------------------
(Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster 
  should have deleted the EKS cluster
  /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:289

• Failure [0.900 seconds]
(Integration) Create, Get, Scale & Delete
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:27
  when creating a cluster with 1 node
  /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:53
    and deleting the cluster
    /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:273
      should have deleted the EKS cluster [It]
      /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:289

      Expected NOT to find a cluster named fabulous-mushroom-1549635872 but it found

      /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:294
------------------------------
(Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster 
  should have deleted the required cloudformation stacks
  /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:297

• Failure [0.786 seconds]
(Integration) Create, Get, Scale & Delete
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:27
  when creating a cluster with 1 node
  /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:53
    and deleting the cluster
    /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:273
      should have deleted the required cloudformation stacks [It]
      /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:297

      Expected NOT to find a Cloudformation stack named eksctl-fabulous-mushroom-1549635872-cluster but it found

      /go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:302
------------------------------


Summarizing 3 Failures:

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should not return an error 
/go/src/github.com/weaveworks/eksctl/integration/common_test.go:65

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should have deleted the EKS cluster 
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:294

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should have deleted the required cloudformation stacks 
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:302

Ran 19 of 19 Specs in 2434.689 seconds
FAIL! -- 16 Passed | 3 Failed | 0 Pending | 0 Skipped
--- FAIL: TestSuite (2434.69s)
FAIL
make: *** [integration-test-container] Error 1
```

I've reverted #513 and run 3 tests at the same time, one succeeded and 2 failed with identical results:

```
Summarizing 5 Failures:

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and add the second nodegroup [It] should not return an error 
/go/src/github.com/weaveworks/eksctl/integration/common_test.go:66

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and add the second nodegroup [It] should make it 8 nodes total 
/go/src/github.com/weaveworks/eksctl/vendor/github.com/dlespiau/kube-test-harness/test.go:200

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should not return an error 
/go/src/github.com/weaveworks/eksctl/integration/common_test.go:65

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should have deleted the EKS cluster 
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:294

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should have deleted the required cloudformation stacks 
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:302

Ran 19 of 19 Specs in 2556.289 seconds
FAIL! -- 14 Passed | 5 Failed | 0 Pending | 0 Skipped
--- FAIL: TestSuite (2556.29s)
FAIL
make: *** [integration-test-container-pre-built] Error 1
```

```
Summarizing 5 Failures:

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and add the second nodegroup [It] should not return an error 
/go/src/github.com/weaveworks/eksctl/integration/common_test.go:66

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and add the second nodegroup [It] should make it 8 nodes total 
/go/src/github.com/weaveworks/eksctl/vendor/github.com/dlespiau/kube-test-harness/test.go:200

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should not return an error 
/go/src/github.com/weaveworks/eksctl/integration/common_test.go:65

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should have deleted the EKS cluster 
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:294

[Fail] (Integration) Create, Get, Scale & Delete when creating a cluster with 1 node and deleting the cluster [It] should have deleted the required cloudformation stacks 
/go/src/github.com/weaveworks/eksctl/integration/creategetdelete_test.go:302

Ran 19 of 19 Specs in 2559.808 seconds
FAIL! -- 14 Passed | 5 Failed | 0 Pending | 0 Skipped
--- FAIL: TestSuite (2559.81s)
FAIL
make: *** [integration-test-container-pre-built] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -80,6 +80,7 @@ integration-test-dev: build build-integration-test ## Run the integration tests 
 	@./eksctl utils write-kubeconfig \
 		--auto-kubeconfig \
 		--name=$(TEST_CLUSTER)
+	$(info it is recommended to watch events with "kubectl get events --watch --all-namespaces --kubeconfig=$(HOME)/.kube/eksctl/clusters/$(TEST_CLUSTER)")
 	@cd integration ; ../eksctl-integration-test -test.timeout 21m \
 		$(INTEGRATION_TEST_ARGS) \
 		-eksctl.cluster=$(TEST_CLUSTER) \

--- a/Makefile
+++ b/Makefile
@@ -60,6 +60,10 @@ integration-test: build build-integration-test ## Run the integration tests (wit
 
 .PHONY: integration-test-container
 integration-test-container: eksctl-image ## Run the integration tests inside a Docker container
+	$(MAKE) integration-test-container-pre-built
+
+.PHONY: integration-test-container-pre-built
+integration-test-container-pre-built: ## Run the integration tests inside a Docker container
 	@docker run \
 	  --env=AWS_PROFILE \
 	  --volume=$(HOME)/.aws:/root/.aws \

--- a/integration/creategetdelete_test.go
+++ b/integration/creategetdelete_test.go
@@ -168,6 +168,9 @@ var _ = Describe("(Integration) Create, Get, Scale & Delete", func() {
 
 				AfterEach(func() {
 					test.Close()
+					Eventually(func() int {
+						return len(test.ListPods(test.Namespace, metav1.ListOptions{}).Items)
+					}, "3m", "1s").Should(BeZero())
 				})
 
 				It("should deploy podinfo service to the cluster and access it via proxy", func() {

